content/client-lib-development-guide/features.textile

```diff
@@ -160,8 +158,7 @@ h3(#rest-channel). Channel
 **** @(RSL1g1b)@ Ably is expected to automatically assign a @clientId@ upon receiving the @Message@. A test should assert via the history API that the @clientId@ value is populated for the @Message@ when received
 *** @(RSL1g2)@ When publishing a @Message@ with the @clientId@ attribute value set to the authenticated client's @clientId@, Ably will accept the message and publish it. A test should assert that the @clientId@ value is populated for the @Message@ when received
 *** @(RSL1g3)@ When publishing a @Message@ with a different @clientId@ attribute value to the authenticated @clientId@, the client library should reject the message, and an exception should be raised. The connection and channel remains available for further operations
-*** @(RSL1g4)@ When using the Realtime library, it is possible to publish a message with an explicit @clientId@ before the library has completed authentication. As such, it is possible that the @clientId@ inferred following authentication is incompatible with the explicit @clientId@ of the message queued for delivery. In this case, it is not expected that the client library will reject the already queued message as the Ably server will reject the message anyway when received. However, all future messages published with an incompatible @clientId@ should be immediately rejected by the client library
-*** @(RSL1g5)@ When using the REST library, publishing a message with an explicit @clientId@ that is incompatible with the @clientId@ inferred following authentication will cause the library to reject the message immediately. Unlike the Realtime library, the message is never queued,and as such, authentication will be completed synchronously before attempting to publish the message over REST
+*** @(RSL1g4)@ When publishing a message with an explicit @clientId@ that is incompatible with the @clientId@ (either inferred or explicitly configured in the token or @ClientOptions@), the library will reject the message immediately and raise an exception.
```

Comment:
Well with all client libraries so far, an exception is raised in the case of REST.  Do you think that is not always going to be the case?  Perhaps not, your wording may be more flexible moving forwards.